projects/dsp-ui/src/lib/search/advanced-search/select-property/specify-property-value/search-link-value/search-link-value.component.ts

```diff
@@ -0,0 +1,130 @@
+import { Component, Inject, Input, OnDestroy, OnInit } from '@angular/core';
+import { IRI, PropertyValue, Value } from '../operator';
+import { Constants, ReadResource, ReadResourceSequence, KnoraApiConnection } from '@dasch-swiss/dsp-js';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { DspApiConnectionToken } from '../../../../../core';
+
+// https://stackoverflow.com/questions/45661010/dynamic-nested-reactive-form-expressionchangedafterithasbeencheckederror
+const resolvedPromise = Promise.resolve(null);
+
+@Component({
+  selector: 'dsp-search-link-value',
+  templateUrl: './search-link-value.component.html',
+  styleUrls: ['./search-link-value.component.scss']
+})
+export class SearchLinkValueComponent implements OnInit, OnDestroy, PropertyValue {
+
+    // parent FormGroup
+    @Input() formGroup: FormGroup;
+
+    type = Constants.LinkValue;
+
+    form: FormGroup;
+
+    resources: ReadResource[];
+
+    private _restrictToResourceClass: string;
+
+    @Input()
+    set restrictResourceClass(value: string) {
+        this._restrictToResourceClass = value;
+    }
+
+    get restrictResourceClass() {
+        return this._restrictToResourceClass;
+    }
+
+    constructor(
+        @Inject(DspApiConnectionToken) private knoraApiConnection: KnoraApiConnection,
+        @Inject(FormBuilder) private fb: FormBuilder) {
+    }
+
+    /**
+     * Displays a selected resource using its label.
+     *
+     * @param resource the resource to be displayed (or no selection yet).
+     */
+    displayResource(resource: ReadResource | null) {
+
+        // null is the initial value (no selection yet)
+        if (resource !== null) {
+            return resource.label;
+        }
+    }
+
+    /**
+     * Search for resources whose labels contain the given search term, restricting to to the given properties object constraint.
+     *
+     * @param searchTerm the term to search for.
+     */
+    searchByLabel(searchTerm: string) {
```

Comment:
how can we test the search by label? I tried with BEOL ontology > Letter > Author is equal to ... but no list of suggestions

Reply:
If you do not have the BEOL data, this won't work. This would search for BEOL persons.

With the test data, you can search for: anything -> Thing -> HasOtherThing -> thing..

I think I will add an e2e test for this.

Reply:
<img width="678" alt="Screen Shot 2020-07-02 at 9 53 07 AM" src="https://user-images.githubusercontent.com/6000023/86331822-e511b980-bc49-11ea-95ca-6d4ead067166.png">

Reply:
yes I found it, it works on my side

Reply:
I added an e2e test for this in https://github.com/dasch-swiss/knora-ui-ng-lib/pull/107/commits/76a9835fdb10ac961d09c813df72aac313f7f84d